* 페이지 진입 > 박스오피스 메뉴
 일별(default)
  - 24년 9/13 start_date 설정하여 일주일치 한번에 
  - 25년 9/26, 10/3, 10/10, 10/17 start_date 설정하여 일주일치 한번에 
  - 각 지역별로 조회
  - 한국/외국 나눠서
 ~~월별 메뉴 이동~~
 ~~연도별 메뉴 이동~~


In [11]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.support.select import Select
import time
import random
from tqdm import tqdm


In [12]:
def set_daily_searh_date_option_by_start_date(driver, start_date):
    # global driver
    # 2.1 시작일 선택
    driver.find_element(By.ID, "sSearchFrom").click()
    # 2.2 년 
    wanna_this_year = start_date[0]#"2025"
    y_select = driver.find_element(By.CSS_SELECTOR,".ui-datepicker-year")
    Select(y_select).select_by_value(wanna_this_year)
    # 2.3 월
    wanna_this_month = start_date[1]#"9월"
    m_select = driver.find_element(By.CSS_SELECTOR,".ui-datepicker-month")
    Select(m_select).select_by_visible_text(wanna_this_month)
    # 2.4 일
    wanna_this_day = start_date[2]#"26"
    d_table = driver.find_element(By.CSS_SELECTOR,".ui-datepicker-calendar")
    d_elems = d_table.find_elements(By.TAG_NAME, "a")
    for d_elem in d_elems:
        # print(d_elem.text)#클릭 이후 찍으면 -> 에러! 왜냐? 알아봐라. 대충은 알겠음.
        if d_elem.text == wanna_this_day:
            # print("! ", d_elem.text)#클릭 이후 찍으면 -> 에러! 왜냐? 알아봐라. 대충은 알겠음.
            d_elem.click()
            break


In [13]:
from selenium.common.exceptions import StaleElementReferenceException

# 조회 클릭
def clickSearchButton():
    print('조회!')
    global driver
    
    try:
        search_btn = driver.find_element(By.CLASS_NAME, "wrap_btn").find_element(By.TAG_NAME, "button")
        WebDriverWait(driver, 240).until(EC.element_to_be_clickable(search_btn))
        search_btn.click()
        
    except StaleElementReferenceException:
        print("StaleElementReferenceException!")
        time.sleep(1)
        search_btn = driver.find_element(By.CLASS_NAME, "wrap_btn").find_element(By.TAG_NAME, "button")
        WebDriverWait(driver, 240).until(EC.element_to_be_clickable(search_btn))
        search_btn.click()

    except TimeoutError as e:
        print("TimeoutException!")
        # time.sleep(60)# 서버에서 짤리면 의미없음. 그리고, 에러 여기로 잡히지 않는 것 같음.
        
    finally:
        pass


    

In [14]:
from httpcore import TimeoutException
from selenium.common.exceptions import StaleElementReferenceException
from selenium.common.exceptions import NoAlertPresentException

def downloadExcelFile():
    print('Excel 다운로드!')
    global driver

    try:
        exl_btn = driver.find_element(By.CSS_SELECTOR,".f_sch").find_element(By.TAG_NAME, "a")
        WebDriverWait(driver, 240).until(EC.element_to_be_clickable(exl_btn))
        exl_btn.click()
    except StaleElementReferenceException:
        print("StaleElementReferenceException!")
        time.sleep(1)
        exl_btn = driver.find_element(By.CSS_SELECTOR,".f_sch").find_element(By.TAG_NAME, "a")
        WebDriverWait(driver, 240).until(EC.element_to_be_clickable(exl_btn))
        exl_btn.click()
    except TimeoutException as e:
        print("TimeoutException!")
        time.sleep(60)
    finally:
        pass

    try:
        alert = driver.switch_to.alert
        alert.accept()
        print("clearExlDownAlert")
    except NoAlertPresentException as e:
        print('NoAlertPresentException!')#, str(e))


In [15]:
# 박스오피스 일별 진입

options = webdriver.ChromeOptions()
download_path = '/home/addinedu/Project_ws/eda-repo-2/Boxoffice/output'
driver_path = '/home/addinedu/Project_ws/eda-repo-2/ShoppingList/driver/chromedriver'
prefs = {'download.default_directory':download_path, 'download.prompt_for_download':False}
options.add_experimental_option('prefs', prefs)

driver = webdriver.Chrome(service=Service(driver_path), options=options)
driver.get("https://www.kobis.or.kr/")

# Top Menu : boxoffice click
top_menu_link_list = driver.find_elements(By.CLASS_NAME, 'tit_nav')
box_office_link = top_menu_link_list[1]
box_office_link.send_keys(Keys.ENTER)

In [16]:
#   - 24년 9/13 start_date 설정하여 일주일치 한번에 
#   - 25년 9/26, 10/3, 10/10, 10/17 start_date 설정하여 일주일치 한번에 
daily_start_date_list = [
    ['2024', '9월', '13'],
    ['2025', '9월', '26'],
    ['2025', '10월', '3'],
    ['2025', '10월', '10'],
    ['2025', '10월', '17']
]

#한국/외국
nation_select = driver.find_element(By.ID, "sRepNationCd")
nation_option_list = nation_select.find_elements(By.TAG_NAME, "option")
nation_list = [n_opt.get_attribute("value") for n_opt in nation_option_list if n_opt.get_attribute("value")]
print('nation_list', nation_list)

#region
#지역 선택: 개별지역 돌아가며
region_select = driver.find_element(By.ID, "sWideAreaCd")
region_option_list = region_select.find_elements(By.TAG_NAME, "option")
region_list = [r_opt.get_attribute("value") for r_opt in region_option_list if r_opt.get_attribute("value")]

#date
date_len = len(daily_start_date_list)
# for di, start_date in enumerate(tqdm(daily_start_date_list)):
for di, start_date in enumerate(daily_start_date_list):
    print(di+1, '/', date_len, start_date, end=' ')
    set_daily_searh_date_option_by_start_date(driver, start_date)
    time.sleep(0.5)

    for n_val in nation_list:
        print(n_val, end='')
        nation_select = driver.find_element(By.ID, "sRepNationCd")
        Select(nation_select).select_by_value(n_val)

        for r_val in tqdm(region_list):#[:3]):#!! 일부만!!
            # print(r_val)
            region_select = driver.find_element(By.ID, "sWideAreaCd")# !!이렇게 안쓰니 StaleElementReferenceException 에러가 나는데...
            Select(region_select).select_by_value(r_val)
            time.sleep( random.uniform(1,2) )

            clickSearchButton() # type: ignore # time.sleep( random.uniform(5,8) )

            downloadExcelFile()
            time.sleep( random.uniform(3,5) )


nation_list ['K', 'F']
1 / 5 ['2024', '9월', '13'] K

  0%|          | 0/17 [00:00<?, ?it/s]

조회!
Excel 다운로드!
clearExlDownAlert


  6%|▌         | 1/17 [00:07<01:57,  7.33s/it]

조회!
Excel 다운로드!
clearExlDownAlert


 12%|█▏        | 2/17 [00:14<01:45,  7.02s/it]


KeyboardInterrupt: 

^^^^ KeyboardInterrupt : 다 받은 후 재시도. 로그 줄이기 위해 강제 종료시킴 ^^^^

In [19]:
# down file check. len
from glob import glob

file_list = glob(download_path + '/KOBIS_*')
len(file_list)


170

In [20]:
import pandas as pd
import re

def get_dataframe_from_daily_excel(soup):
    div_elems = soup.find_all('div')
    table_elems = soup.find_all('table')[1:]#날짜별로, 첫 테이블은 검색조건

    total_data_list = []

    for ti, table in enumerate(table_elems):
        date_div = div_elems[ti]
        date = re.findall(r'\d+년 \d+월 \d+일', date_div.text.strip())[0]# print(date_div.text.strip())

        rows = table.find_all('tr')
        end_row_num = len(rows)
        for ri,row in enumerate(rows):

            row_data = []

            if ri==0:#head
                if ti == 0:
                    row_data.append('일자')
                    cols = row.find_all('th')
                    for ci, th in enumerate(cols):
                        if ci in (1,2,3,8,12,13,14,16,18,19,20,21):#영화명, 개봉일, 매출액, 관객수, 스크린수, 상영횟수, 14대표국적, 16제작사, 17배급사, 18등급, 19장르, 20감독, 21배우
                            row_data.append(th.text.strip())

                    total_data_list.append(row_data)

            elif ri == end_row_num-1:#bottom sum
                pass

            else:
                row_data.append(date)
                cols = row.find_all('td')

                for ci, td in enumerate(cols):
                    # if ci == 0:
                    #     row_data.append(int(td.text.strip()))
                    # elif ci == 1:#str 영화명
                    #     row_data.append(td.text.strip())
                    # elif ci == 3:
                    #     row_data.append(int(td.text.strip()))
                    # elif ci == 8:
                    #     row_data.append(int(td.text.strip()))
                    # elif ci == 12:
                    #     row_data.append(int(td.text.strip()))
                    # elif ci == 13:
                    #     row_data.append(int(td.text.strip()))
                    if ci in (1, 14,16,18,19,20,21):
                        row_data.append(td.text.strip())
                    elif ci == 2:
                        row_data.append(td.text.strip())
                    elif ci in (3,8,12,13):
                        row_data.append(int(td.text.strip()))



                total_data_list.append(row_data)

    df = pd.DataFrame(total_data_list[1:], columns=total_data_list[0])

    return df

In [21]:
from bs4 import BeautifulSoup as bs
import pandas as pd

total_df = pd.DataFrame()

for idx, file in enumerate(tqdm(file_list)):
    print(idx, file, end=", ")

    with open(file, 'r') as html:
        html_doc = html.read()
        soup = bs(html_doc,'html.parser')

        # 지역명 확인
        search_option_str = soup.find_all('table')[0].text
        temp_str = re.compile("지역:+.[가-힣]+").search(search_option_str)
        region_name = temp_str.group().replace('지역:','').strip()
        
        print(region_name)#, end=", ")

        df = get_dataframe_from_daily_excel(soup)
        # 지역 col 추가: 모든 열에 같은 값으로
        df['지역'] = region_name
        
        total_df = pd.concat([total_df, df])



  0%|          | 0/170 [00:00<?, ?it/s]

0 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192958.218.xls, 

  1%|          | 1/170 [00:00<00:48,  3.52it/s]

서울시
1 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192604.711.xls, 

  2%|▏         | 3/170 [00:00<00:30,  5.53it/s]

충청남도
2 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (4).xls, 충청남도
3 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193120.791.xls, 울산시
4 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (18).xls, 

  3%|▎         | 5/170 [00:01<00:37,  4.40it/s]

경기도
5 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192540.278.xls, 

  5%|▍         | 8/170 [00:01<00:32,  5.05it/s]

서울시
6 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192343.908.xls, 세종시
7 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (37).xls, 충청북도
8 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (80).xls, 

  6%|▌         | 10/170 [00:01<00:26,  6.07it/s]

대전시
9 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (40).xls, 경상남도
10 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (13).xls, 

  7%|▋         | 12/170 [00:02<00:25,  6.10it/s]

울산시
11 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (71).xls, 충청북도
12 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (21).xls, 

  8%|▊         | 13/170 [00:02<00:26,  5.89it/s]

충청남도
13 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (61).xls, 

  9%|▉         | 15/170 [00:02<00:27,  5.67it/s]

부산시
14 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (7).xls, 전라북도
15 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (95).xls, 

  9%|▉         | 16/170 [00:03<00:28,  5.46it/s]

부산시
16 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (11).xls, 

 11%|█         | 18/170 [00:03<00:27,  5.51it/s]

대구시
17 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (36).xls, 강원도
18 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193108.965.xls, 

 11%|█         | 19/170 [00:03<00:23,  6.30it/s]

대구시
19 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (68).xls, 

 12%|█▏        | 20/170 [00:03<00:32,  4.67it/s]

서울시
20 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192629.303.xls, 전라남도
21 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (17).xls, 

 13%|█▎        | 22/170 [00:04<00:40,  3.62it/s]

서울시
22 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192708.144.xls, 

 14%|█▎        | 23/170 [00:04<00:37,  3.88it/s]

인천시
23 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192501.205.xls, 

 15%|█▍        | 25/170 [00:05<00:32,  4.42it/s]

대구시
24 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192436.115.xls, 전라북도
25 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (34).xls, 

 15%|█▌        | 26/170 [00:05<00:37,  3.81it/s]

서울시
26 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192635.460.xls, 제주도
27 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (62).xls, 

 16%|█▋        | 28/170 [00:05<00:28,  4.96it/s]

대구시
28 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (84).xls, 세종시
29 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192454.296.xls, 

 18%|█▊        | 31/170 [00:06<00:25,  5.51it/s]

부산시
30 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (30).xls, 울산시
31 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (1).xls, 

 19%|█▉        | 32/170 [00:06<00:24,  5.74it/s]

경기도
32 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (2).xls, 

 20%|██        | 34/170 [00:06<00:24,  5.59it/s]

강원도
33 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (79).xls, 대구시
34 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192854.895.xls, 

 21%|██        | 36/170 [00:07<00:18,  7.21it/s]

울산시
35 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (26).xls, 제주도
36 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (75).xls, 

 22%|██▏       | 38/170 [00:07<00:19,  6.86it/s]

전라북도
37 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (59).xls, 전라남도
38 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (33).xls, 세종시
39 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (65).xls, 

 24%|██▎       | 40/170 [00:07<00:18,  6.87it/s]

인천시
40 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (91).xls, 

 25%|██▍       | 42/170 [00:08<00:21,  6.02it/s]

경상남도
41 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (23).xls, 경상남도
42 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (78).xls, 

 25%|██▌       | 43/170 [00:08<00:21,  6.04it/s]

부산시
43 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (25).xls, 

 26%|██▋       | 45/170 [00:08<00:22,  5.56it/s]

전라남도
44 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192655.677.xls, 대전시
45 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (100).xls, 

 27%|██▋       | 46/170 [00:08<00:21,  5.74it/s]

광주시
46 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (29).xls, 

 28%|██▊       | 47/170 [00:09<00:27,  4.52it/s]

대전시
47 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192728.310.xls, 

 28%|██▊       | 48/170 [00:09<00:26,  4.56it/s]

서울시
48 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (92).xls, 

 30%|███       | 51/170 [00:09<00:20,  5.73it/s]

전라북도
49 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192418.280.xls, 충청남도
50 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192754.526.xls, 충청남도
51 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192545.965.xls, 

 31%|███       | 53/170 [00:10<00:22,  5.28it/s]

경기도
52 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193042.611.xls, 전라북도
53 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (83).xls, 

 32%|███▏      | 54/170 [00:10<00:19,  5.87it/s]

광주시
54 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (48).xls, 

 34%|███▎      | 57/170 [00:10<00:16,  6.93it/s]

인천시
55 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (77).xls, 제주도
56 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193049.346.xls, 전라남도
57 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (54).xls, 

 34%|███▍      | 58/170 [00:11<00:15,  7.05it/s]

충청북도
58 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (67).xls, 세종시
59 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193029.021.xls, 

 36%|███▌      | 61/170 [00:11<00:14,  7.53it/s]

경상북도
60 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (14).xls, 인천시
61 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (49).xls, 

 37%|███▋      | 63/170 [00:11<00:13,  8.07it/s]

광주시
62 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192411.645.xls, 충청북도
63 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192835.727.xls, 

 38%|███▊      | 65/170 [00:12<00:17,  6.07it/s]

부산시
64 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22.xls, 서울시
65 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (56).xls, 

 39%|███▉      | 66/170 [00:12<00:16,  6.40it/s]

경상북도
66 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192424.202.xls, 

 40%|████      | 68/170 [00:12<00:18,  5.59it/s]

경상북도
67 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (58).xls, 전라북도
68 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (47).xls, 

 41%|████      | 69/170 [00:12<00:16,  6.26it/s]

울산시
69 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (99).xls, 

 42%|████▏     | 71/170 [00:13<00:20,  4.84it/s]

인천시
70 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (66).xls, 광주시
71 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (15).xls, 

 42%|████▏     | 72/170 [00:13<00:18,  5.32it/s]

광주시
72 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192938.049.xls, 

 44%|████▎     | 74/170 [00:13<00:17,  5.54it/s]

인천시
73 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (39).xls, 경상북도
74 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (53).xls, 

 44%|████▍     | 75/170 [00:13<00:16,  5.63it/s]

강원도
75 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (97).xls, 

 45%|████▌     | 77/170 [00:14<00:18,  5.03it/s]

대전시
76 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (32).xls, 광주시
77 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (51).xls, 

 46%|████▋     | 79/170 [00:15<00:24,  3.74it/s]

서울시
78 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (19).xls, 강원도
79 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (22).xls, 

 47%|████▋     | 80/170 [00:15<00:20,  4.38it/s]

경상북도
80 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (3).xls, 

 48%|████▊     | 82/170 [00:15<00:19,  4.55it/s]

충청북도
81 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192648.584.xls, 대구시
82 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (55).xls, 

 49%|████▉     | 84/170 [00:16<00:17,  4.82it/s]

충청남도
83 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (8).xls, 전라남도
84 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (94).xls, 제주도


 51%|█████     | 86/170 [00:16<00:13,  6.27it/s]

85 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (64).xls, 울산시
86 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192531.441.xls, 세종시
87 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (98).xls, 

 52%|█████▏    | 89/170 [00:16<00:12,  6.29it/s]

울산시
88 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (57).xls, 경상남도
89 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192943.468.xls, 

 54%|█████▎    | 91/170 [00:17<00:11,  6.81it/s]

광주시
90 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (89).xls, 충청남도
91 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193101.313.xls, 

 55%|█████▍    | 93/170 [00:17<00:13,  5.88it/s]

부산시
92 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (44).xls, 부산시
93 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (6).xls, 

 55%|█████▌    | 94/170 [00:17<00:11,  6.54it/s]

경상남도
94 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (70).xls, 

 56%|█████▋    | 96/170 [00:17<00:11,  6.40it/s]

강원도
95 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192611.092.xls, 경상북도
96 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (9).xls, 제주도


 58%|█████▊    | 98/170 [00:18<00:09,  7.57it/s]

97 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (74).xls, 경상남도
98 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (69).xls, 

 59%|█████▉    | 100/170 [00:18<00:10,  6.44it/s]

경기도
99 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192701.368.xls, 울산시
100 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192352.521.xls, 

 59%|█████▉    | 101/170 [00:18<00:12,  5.52it/s]

서울시
101 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (20).xls, 

 61%|██████    | 103/170 [00:19<00:12,  5.42it/s]

충청북도
102 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192841.295.xls, 대구시
103 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192507.627.xls, 

 61%|██████    | 104/170 [00:19<00:11,  5.95it/s]

대전시
104 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192429.709.xls, 

 62%|██████▏   | 105/170 [00:19<00:12,  5.19it/s]

경상남도
105 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (31).xls, 

 62%|██████▏   | 106/170 [00:19<00:13,  4.69it/s]

인천시
106 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193009.936.xls, 강원도
107 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192847.610.xls, 

 64%|██████▎   | 108/170 [00:20<00:11,  5.26it/s]

대전시
108 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193016.375.xls, 충청북도
109 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (87).xls, 

 65%|██████▍   | 110/170 [00:20<00:10,  5.86it/s]

강원도
110 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (76).xls, 전라남도
111 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (60).xls, 

 66%|██████▋   | 113/170 [00:20<00:08,  6.55it/s]

제주도
112 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (81).xls, 울산시
113 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192622.389.xls, 

 68%|██████▊   | 115/170 [00:21<00:07,  6.91it/s]

전라북도
114 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (88).xls, 충청북도
115 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192524.796.xls, 

 68%|██████▊   | 116/170 [00:21<00:09,  5.74it/s]

광주시
116 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192641.783.xls, 

 69%|██████▉   | 118/170 [00:21<00:09,  5.67it/s]

부산시
117 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192553.278.xls, 강원도
118 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192513.596.xls, 

 71%|███████   | 120/170 [00:22<00:08,  6.05it/s]

울산시
119 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192441.501.xls, 전라남도
120 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192950.505.xls, 세종시
121 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (96).xls, 

 72%|███████▏  | 122/170 [00:22<00:07,  6.73it/s]

대구시
122 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (93).xls, 

 72%|███████▏  | 123/170 [00:22<00:07,  6.06it/s]

전라남도
123 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193036.055.xls, 경상남도
124 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192405.337.xls, 강원도


 75%|███████▍  | 127/170 [00:22<00:05,  8.16it/s]

125 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192816.180.xls, 전라북도
126 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192801.786.xls, 경상북도
127 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (86).xls, 

 76%|███████▌  | 129/170 [00:23<00:06,  6.20it/s]

경기도
128 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192558.427.xls, 충청북도
129 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (38).xls, 

 76%|███████▋  | 130/170 [00:23<00:05,  6.79it/s]

충청남도
130 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192616.517.xls, 

 78%|███████▊  | 132/170 [00:23<00:06,  6.03it/s]

경상남도
131 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193127.519.xls, 인천시
132 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192809.205.xls, 

 78%|███████▊  | 133/170 [00:24<00:05,  6.72it/s]

경상남도
133 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192748.213.xls, 충청북도
134 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192735.217.xls, 

 79%|███████▉  | 135/170 [00:24<00:05,  6.13it/s]

경기도
135 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192448.520.xls, 제주도
136 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193022.467.xls, 충청남도


 81%|████████  | 138/170 [00:24<00:04,  7.02it/s]

137 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (82).xls, 인천시
138 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192359.193.xls, 

 82%|████████▏ | 139/170 [00:25<00:05,  5.31it/s]

경기도
139 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (85).xls, 

 82%|████████▏ | 140/170 [00:25<00:08,  3.52it/s]

서울시
140 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (27).xls, 

 83%|████████▎ | 141/170 [00:26<00:09,  3.15it/s]

부산시
141 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193139.788.xls, 세종시
142 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (35).xls, 

 85%|████████▍ | 144/170 [00:26<00:05,  4.77it/s]

경기도
143 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (41).xls, 전라북도
144 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (46).xls, 

 86%|████████▌ | 146/170 [00:26<00:05,  4.80it/s]

대전시
145 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (90).xls, 경상북도
146 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (16).xls, 

 87%|████████▋ | 148/170 [00:27<00:03,  6.42it/s]

세종시
147 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (72).xls, 충청남도
148 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193055.564.xls, 제주도
149 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192720.205.xls, 

 89%|████████▉ | 151/170 [00:27<00:02,  7.02it/s]

세종시
150 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192741.022.xls, 강원도
151 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (52).xls, 

 89%|████████▉ | 152/170 [00:27<00:03,  5.70it/s]

경기도
152 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193004.378.xls, 

 91%|█████████ | 154/170 [00:28<00:02,  5.62it/s]

경기도
153 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193114.672.xls, 대전시
154 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (5).xls, 

 92%|█████████▏| 156/170 [00:28<00:02,  6.60it/s]

경상북도
155 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T193132.681.xls, 광주시
156 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192827.530.xls, 

 93%|█████████▎| 158/170 [00:28<00:01,  6.58it/s]

제주도
157 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (42).xls, 전라남도
158 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192714.213.xls, 

 94%|█████████▎| 159/170 [00:28<00:01,  6.27it/s]

광주시
159 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (28).xls, 

 94%|█████████▍| 160/170 [00:29<00:02,  4.73it/s]

대구시
160 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (43).xls, 제주도
161 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (12).xls, 

 96%|█████████▌| 163/170 [00:29<00:01,  6.06it/s]

대전시
162 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192519.036.xls, 인천시
163 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (24).xls, 

 97%|█████████▋| 165/170 [00:30<00:00,  5.57it/s]

전라북도
164 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (73).xls, 경상북도
165 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (50).xls, 

 98%|█████████▊| 167/170 [00:30<00:00,  6.68it/s]

세종시
166 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (45).xls, 대구시
167 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 - 2025-10-22T192822.160.xls, 

 99%|█████████▉| 169/170 [00:30<00:00,  6.30it/s]

전라남도
168 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (10).xls, 부산시
169 /home/addinedu/Project_ws/eda-repo-2/Boxoffice/output/KOBIS_일별_박스오피스_2025-10-22 (63).xls, 

100%|██████████| 170/170 [00:30<00:00,  5.52it/s]

대전시


In [22]:
total_df

일자               영화명         개봉일       매출액   관객수  스크린수  상영횟수  \
0    2025년 10월 21일    극장판 체인소 맨: 레제편  2025-09-24  83787060  7650   134   493   
1    2025년 10월 21일      원 배틀 애프터 어나더  2025-10-01  20112040  1931   113   217   
2    2025년 10월 21일  극장판 귀멸의 칼날: 무한성편  2025-08-22  14885500  1405    78   129   
3    2025년 10월 21일   극장판 주술회전: 회옥·옥절  2025-10-16  13826100  1320    99   209   
4    2025년 10월 21일                웨폰  2025-10-15  12750600  1286    85   152   
..             ...               ...         ...       ...   ...   ...   ...   
162  2025년 09월 26일            결혼 피로연  2025-09-24     21000     3     1     2   
163  2025년 09월 26일     베르메르: 위대한 전시회  2025-09-17      8000     1     1     1   
164  2025년 09월 26일                부재  2025-08-27      8000     1     1     1   
165  2025년 09월 26일         내 말 좀 들어줘  2025-08-20     10000     1     1     1   
166  2025년 09월 26일               투게더  2025-09-03     14000     1     1     1   

    대표국적 제작사        등급             장르          감독  \
0     일본      15세이상관람가  애니메이션,액션,어드벤처               
1     미국      15세이상관람가          범죄,액션   폴 토마스 앤더슨   
2     일본      15세이상관람가          애니메이션    소토자키 하루오   
3     일본      15세이상관람가          애니메이션     고쇼조노 쇼타   
4     미국       청소년관람불가    공포(호러),미스터리       잭 크레거   
..   ...  ..       ...            ...         ...   
162   미국      15세이상관람가     코미디,멜로/로맨스       앤드류 안   
163   영국         전체관람가          다큐멘터리  데이비드 비커스태프   
164   중국      12세이상관람가            드라마               
165   영국      12세이상관람가            드라마       마이크 리   
166   미국      15세이상관람가  공포(호러),멜로/로맨스     마이클 쉥크스   

                                                    배우   지역  
0                                                       서울시  
1     레오나르도 디카프리오,숀 펜,베니시오 델 토로,레지나 홀,테야나 테일러,체이스 인피니티  서울시  
2    하나에 나츠키,키토 아카리,시모노 히로,마츠오카 요시츠구,오카모토 노부히코,사쿠라이...  서울시  
3                                                       서울시  
4    조슈 브롤린,줄리아 가너,올든 에런라이크,오스틴 아브람스,크리스토퍼 캐리,베네딕트 ...  서울시  
..                                                 ...  ...  
162                윤여정,한기찬,보웬 양,릴리 글래드스톤,켈리 마리 트란,조안 첸  대전시  
163                                                     대전시  
164                                                     대전시  
165                                         마리안 장 밥티스트  대전시  
166                                     데이브 프랭코,알리슨 브리  대전시  

[21216 rows x 14 columns]

In [23]:
total_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 21216 entries, 0 to 166
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   일자      21216 non-null  object
 1   영화명     21216 non-null  object
 2   개봉일     21216 non-null  object
 3   매출액     21216 non-null  int64 
 4   관객수     21216 non-null  int64 
 5   스크린수    21216 non-null  int64 
 6   상영횟수    21216 non-null  int64 
 7   대표국적    21216 non-null  object
 8   제작사     21216 non-null  object
 9   등급      21216 non-null  object
 10  장르      21216 non-null  object
 11  감독      21216 non-null  object
 12  배우      21216 non-null  object
 13  지역      21216 non-null  object
dtypes: int64(4), object(10)
memory usage: 2.4+ MB


## 데이터 전처리
1. datetime: 일자(관객이 영화본날), 개봉일
 - 일자 형식  : 2025년 10월 21일
 - 개봉일 형식 : 2025-09-24
2. int : 매출액, 관객수, 스크린수, 상영횟수
3. object(문자열) : 그외 컬럼

In [24]:
from datetime import datetime

date = datetime.strptime('2024-01-20', '%Y-%m-%d')
date


datetime.datetime(2024, 1, 20, 0, 0)

In [25]:
import pandas as pd

# 형식: "%Y-%m-%d"
total_df['개봉일'] = pd.to_datetime(total_df['개봉일'])
total_df['개봉일']

0     2025-09-24
1     2025-10-01
2     2025-08-22
3     2025-10-16
4     2025-10-15
         ...    
162   2025-09-24
163   2025-09-17
164   2025-08-27
165   2025-08-20
166   2025-09-03
Name: 개봉일, Length: 21216, dtype: datetime64[ns]

In [26]:
import pandas as pd

'''
# 형식: "%Y년 %m월 %d일"
# total_df['일자'] = pd.to_datetime(total_df['일자'])
# total_df['일자']

'raise' (기본값)	변환할 수 없는 값이 있으면 ValueError와 같은 오류를 발생시킵니다.
'coerce'	변환할 수 없는 값을 NaT (Not a Time) 값으로 바꿉니다.
'ignore'	변환할 수 없는 값이 있으면 해당 값을 그대로 반환합니다.
'''

total_df['일자'] = pd.to_datetime(total_df['일자'], format="%Y년 %m월 %d일", errors='coerce')
total_df['일자']

0     2025-10-21
1     2025-10-21
2     2025-10-21
3     2025-10-21
4     2025-10-21
         ...    
162   2025-09-26
163   2025-09-26
164   2025-09-26
165   2025-09-26
166   2025-09-26
Name: 일자, Length: 21216, dtype: datetime64[ns]

In [27]:
total_df['일자'].unique()

<DatetimeArray>
['2025-10-21 00:00:00', '2025-10-20 00:00:00', '2025-10-19 00:00:00',
 '2025-10-18 00:00:00', '2025-10-17 00:00:00', '2025-10-16 00:00:00',
 '2025-10-15 00:00:00', '2025-10-14 00:00:00', '2025-10-13 00:00:00',
 '2025-10-12 00:00:00', '2025-10-11 00:00:00', '2025-10-10 00:00:00',
 '2024-09-19 00:00:00', '2024-09-18 00:00:00', '2024-09-17 00:00:00',
 '2024-09-16 00:00:00', '2024-09-15 00:00:00', '2024-09-14 00:00:00',
 '2024-09-13 00:00:00', '2025-10-09 00:00:00', '2025-10-08 00:00:00',
 '2025-10-07 00:00:00', '2025-10-06 00:00:00', '2025-10-05 00:00:00',
 '2025-10-04 00:00:00', '2025-10-03 00:00:00', '2025-10-02 00:00:00',
 '2025-10-01 00:00:00', '2025-09-30 00:00:00', '2025-09-29 00:00:00',
 '2025-09-28 00:00:00', '2025-09-27 00:00:00', '2025-09-26 00:00:00']
Length: 33, dtype: datetime64[ns]

### 날짜가 뒤죽박죽이다. 정렬을 해보자.

In [28]:
# Sorting: 일자 asc, 지역 asc
total_df = total_df.sort_values(by=['일자', '지역', '매출액'], ascending=[True, True, True])
total_df

일자               영화명        개봉일      매출액  관객수  스크린수  상영횟수 대표국적  \
115 2024-09-13  죽고 싶지만 사랑은 하고 싶어 2024-09-04     6000    1     1     1   미국   
116 2024-09-13        래시: 뉴 어드벤처 2024-09-12     6000    1     1     1   독일   
117 2024-09-13  극장총집편 봇치 더 록! 전편 2024-08-07    13000    1     1     1   일본   
101 2024-09-13              늘봄가든 2024-08-21    14000    2     1     1   한국   
114 2024-09-13              주토피아 2016-02-17    18000    2     1     1   미국   
..         ...               ...        ...      ...  ...   ...   ...  ...   
2   2025-10-21  극장판 귀멸의 칼날: 무한성편 2025-08-22  1194360  117    16    22   일본   
2   2025-10-21            어쩔수가없다 2025-09-24  1828700  211    24    52   한국   
1   2025-10-21             연의 편지 2025-10-01  2104800  304    13    14   한국   
0   2025-10-21                보스 2025-10-03  3714400  510    28    84   한국   
0   2025-10-21    극장판 체인소 맨: 레제편 2025-09-24  5212970  551    29    99   일본   

                        제작사        등급             장르             감독  \
115                          12세이상관람가     드라마,멜로/로맨스        레이첼 램버트   
116                             전체관람가    어드벤처,가족,드라마                  
117                          12세이상관람가          애니메이션      사이토 케이이치로   
101              (주)바이어스이엔티  15세이상관람가     공포(호러),스릴러            구태진   
114                             전체관람가  애니메이션,액션,어드벤처  바이론 하워드,리치 무어   
..                      ...       ...            ...            ...   
2                            15세이상관람가          애니메이션       소토자키 하루오   
2    모호필름,씨제이이엔엠 스튜디오스 주식회사  15세이상관람가        스릴러,코미디            박찬욱   
1      스튜디오리코 주식회사,스튜디오엔(주)  12세이상관람가          애니메이션            김용환   
0               (주)하이브미디어코프  15세이상관람가         코미디,액션            라희찬   
0                            15세이상관람가  애니메이션,액션,어드벤처                  

                                                    배우    지역  
115                                   데이지 리들리,데이브 메르헤예   강원도  
116                                   니코 마리츠카,카타리나 슈틀러   강원도  
117                  아오야마 요시노,스즈시로 사유미,미즈노 사쿠,하세가와 이쿠미   강원도  
101                                조윤희,이지오,김주령,허동원,정인겸   강원도  
114                                지니퍼 굿윈,제이슨 베이트먼,샤키라   강원도  
..                                                 ...   ...  
2    하나에 나츠키,키토 아카리,시모노 히로,마츠오카 요시츠구,오카모토 노부히코,사쿠라이...  충청북도  
2                      이병헌,손예진,박희순,이성민,염혜란,차승원,유연석,이진결  충청북도  
1                                      이수현,김민주,민승우,남도형  충청북도  
0                 조우진,정경호,박지환,이규형,오달수,황우슬혜,정유진,고창석,이성민  충청북도  
0                                                       충청북도  

[21216 rows x 14 columns]

In [30]:
# 정렬했으니, reindexing
total_df.reset_index(drop=True, inplace=True)

In [31]:
total_df

일자               영화명        개봉일      매출액  관객수  스크린수  상영횟수 대표국적  \
0     2024-09-13  죽고 싶지만 사랑은 하고 싶어 2024-09-04     6000    1     1     1   미국   
1     2024-09-13        래시: 뉴 어드벤처 2024-09-12     6000    1     1     1   독일   
2     2024-09-13  극장총집편 봇치 더 록! 전편 2024-08-07    13000    1     1     1   일본   
3     2024-09-13              늘봄가든 2024-08-21    14000    2     1     1   한국   
4     2024-09-13              주토피아 2016-02-17    18000    2     1     1   미국   
...          ...               ...        ...      ...  ...   ...   ...  ...   
21211 2025-10-21  극장판 귀멸의 칼날: 무한성편 2025-08-22  1194360  117    16    22   일본   
21212 2025-10-21            어쩔수가없다 2025-09-24  1828700  211    24    52   한국   
21213 2025-10-21             연의 편지 2025-10-01  2104800  304    13    14   한국   
21214 2025-10-21                보스 2025-10-03  3714400  510    28    84   한국   
21215 2025-10-21    극장판 체인소 맨: 레제편 2025-09-24  5212970  551    29    99   일본   

                          제작사        등급             장르             감독  \
0                              12세이상관람가     드라마,멜로/로맨스        레이첼 램버트   
1                                 전체관람가    어드벤처,가족,드라마                  
2                              12세이상관람가          애니메이션      사이토 케이이치로   
3                  (주)바이어스이엔티  15세이상관람가     공포(호러),스릴러            구태진   
4                                 전체관람가  애니메이션,액션,어드벤처  바이론 하워드,리치 무어   
...                       ...       ...            ...            ...   
21211                          15세이상관람가          애니메이션       소토자키 하루오   
21212  모호필름,씨제이이엔엠 스튜디오스 주식회사  15세이상관람가        스릴러,코미디            박찬욱   
21213    스튜디오리코 주식회사,스튜디오엔(주)  12세이상관람가          애니메이션            김용환   
21214             (주)하이브미디어코프  15세이상관람가         코미디,액션            라희찬   
21215                          15세이상관람가  애니메이션,액션,어드벤처                  

                                                      배우    지역  
0                                       데이지 리들리,데이브 메르헤예   강원도  
1                                       니코 마리츠카,카타리나 슈틀러   강원도  
2                      아오야마 요시노,스즈시로 사유미,미즈노 사쿠,하세가와 이쿠미   강원도  
3                                    조윤희,이지오,김주령,허동원,정인겸   강원도  
4                                    지니퍼 굿윈,제이슨 베이트먼,샤키라   강원도  
...                                                  ...   ...  
21211  하나에 나츠키,키토 아카리,시모노 히로,마츠오카 요시츠구,오카모토 노부히코,사쿠라이...  충청북도  
21212                    이병헌,손예진,박희순,이성민,염혜란,차승원,유연석,이진결  충청북도  
21213                                    이수현,김민주,민승우,남도형  충청북도  
21214               조우진,정경호,박지환,이규형,오달수,황우슬혜,정유진,고창석,이성민  충청북도  
21215                                                     충청북도  

[21216 rows x 14 columns]

In [33]:
# to CSV file
total_df.to_csv(download_path + '/kobis_daily_2024_2025_chuseok.csv', sep="|", encoding='utf-8')

In [34]:
# to json file
total_df.to_json(download_path + '/kobis_daily_2024_2025_chuseok.json', indent=2, force_ascii=False)

## !! daily OK !!

### 이제 DB에 데이터 입력한다!

In [39]:
total_df['대표국적'].unique()
len(total_df['대표국적'].unique())

30

In [40]:
#column 순서 정리해야
# 지역 column -> 일자 뒤로 
total_df.columns
total_df = total_df[['일자', '지역', '영화명', '개봉일', '관객수', '스크린수', '상영횟수', '매출액', '대표국적', '제작사', '등급',
       '장르', '감독', '배우']]

### Table 정규화. movie는...
1. boxoffice : ['관람일자', '영화명', '개봉일', '매출액', '관객수', '스크린수', '상영횟수'] #일자 -> 관람일자
2. movie : ['영화명', '개봉일', '대표국적', '제작사', '등급', '장르', '감독', '배우']

In [ ]:
import numpy as np

## total_df is_korean 컬럼 추가 : 대표국적 컬럼 이용해서
total_df['is_korean'] = np.where(total_df['대표국적'] == '한국', 'k', 'f')
total_df

일자    지역               영화명        개봉일  관객수  스크린수  상영횟수      매출액  \
0     2024-09-13   강원도  죽고 싶지만 사랑은 하고 싶어 2024-09-04    1     1     1     6000   
1     2024-09-13   강원도        래시: 뉴 어드벤처 2024-09-12    1     1     1     6000   
2     2024-09-13   강원도  극장총집편 봇치 더 록! 전편 2024-08-07    1     1     1    13000   
3     2024-09-13   강원도              늘봄가든 2024-08-21    2     1     1    14000   
4     2024-09-13   강원도              주토피아 2016-02-17    2     1     1    18000   
...          ...   ...               ...        ...  ...   ...   ...      ...   
21211 2025-10-21  충청북도  극장판 귀멸의 칼날: 무한성편 2025-08-22  117    16    22  1194360   
21212 2025-10-21  충청북도            어쩔수가없다 2025-09-24  211    24    52  1828700   
21213 2025-10-21  충청북도             연의 편지 2025-10-01  304    13    14  2104800   
21214 2025-10-21  충청북도                보스 2025-10-03  510    28    84  3714400   
21215 2025-10-21  충청북도    극장판 체인소 맨: 레제편 2025-09-24  551    29    99  5212970   

      대표국적                     제작사        등급             장르             감독  \
0       미국                          12세이상관람가     드라마,멜로/로맨스        레이첼 램버트   
1       독일                             전체관람가    어드벤처,가족,드라마                  
2       일본                          12세이상관람가          애니메이션      사이토 케이이치로   
3       한국              (주)바이어스이엔티  15세이상관람가     공포(호러),스릴러            구태진   
4       미국                             전체관람가  애니메이션,액션,어드벤처  바이론 하워드,리치 무어   
...    ...                     ...       ...            ...            ...   
21211   일본                          15세이상관람가          애니메이션       소토자키 하루오   
21212   한국  모호필름,씨제이이엔엠 스튜디오스 주식회사  15세이상관람가        스릴러,코미디            박찬욱   
21213   한국    스튜디오리코 주식회사,스튜디오엔(주)  12세이상관람가          애니메이션            김용환   
21214   한국             (주)하이브미디어코프  15세이상관람가         코미디,액션            라희찬   
21215   일본                          15세이상관람가  애니메이션,액션,어드벤처                  

                                                      배우 is_korean  
0                                       데이지 리들리,데이브 메르헤예         f  
1                                       니코 마리츠카,카타리나 슈틀러         f  
2                      아오야마 요시노,스즈시로 사유미,미즈노 사쿠,하세가와 이쿠미         f  
3                                    조윤희,이지오,김주령,허동원,정인겸         k  
4                                    지니퍼 굿윈,제이슨 베이트먼,샤키라         f  
...                                                  ...       ...  
21211  하나에 나츠키,키토 아카리,시모노 히로,마츠오카 요시츠구,오카모토 노부히코,사쿠라이...         f  
21212                    이병헌,손예진,박희순,이성민,염혜란,차승원,유연석,이진결         k  
21213                                    이수현,김민주,민승우,남도형         k  
21214               조우진,정경호,박지환,이규형,오달수,황우슬혜,정유진,고창석,이성민         k  
21215                                                            f  

[21216 rows x 15 columns]

In [46]:
# dt_id 추가 
# 형식 yymmdd99
total_df['dt_id'] = total_df['일자'].dt.strftime('%y%m%d99')
total_df['dt_id'].unique()


array(['24091399', '24091499', '24091599', '24091699', '24091799',
       '24091899', '24091999', '25092699', '25092799', '25092899',
       '25092999', '25093099', '25100199', '25100299', '25100399',
       '25100499', '25100599', '25100699', '25100799', '25100899',
       '25100999', '25101099', '25101199', '25101299', '25101399',
       '25101499', '25101599', '25101699', '25101799', '25101899',
       '25101999', '25102099', '25102199'], dtype=object)

In [45]:
# total_df.columns.unique()
total_df['is_korean'].unique()

array(['f', 'k'], dtype=object)

In [ ]:
import mysql.connector
import pandas as pd
from mysql.connector import Error

try:
    conn = mysql.connector.connect(
        host='localhost',
        user='root',
        password='1234',
        database='amrbase'
    )
    cursor = conn.cursor()

    for i, row in total_df.iterrows():
        if i == 0:
            continue

        print(i, row)#, end='')

        cursor.execute("""
            INSERT INTO boxoffice (
                            movie_name, release_date, dt_id, region
                            , income, views, screen, play_count, is_korean
                       )                
            VALUES (%s,%s,%s,%s,%s,%s,%s,%s,%s)
        """, (
            row['영화명'], row['개봉일'], row['dt_id'], row['지역']
            , row['매출액'], row['관객수'], row['스크린수'], row['상영횟수']
            , row['is_korean']
        ))
    
    conn.commit()
    print("MySQL로 데이터 업로드 OK!")

except Error as e:
    print(f"Error ! {e}")
    pass

finally:
    conn.close()



<<<------------여기까지 Daily boxoffice crawling 완료 ------------>>> 

In [20]:
# left menu : 트리구조로 하위 메뉴들이 있다.
# '월별' 클릭 ('연도별', '일별')
def left_date_menu_change(sub_menu_name):
    '''
    sub_menu_name:'연도별', '월별', '일별'
    '''
    global driver

    left_menu = driver.find_element(By.CLASS_NAME, "list_snb")
    box_office_li = left_menu.find_element(By.CSS_SELECTOR, "li")
    sub_list = box_office_li.find_elements(By.CSS_SELECTOR, "a")
    for a in sub_list:
        # print(a.text)
        if a.text == sub_menu_name:
            a.click()
            break

In [62]:
left_date_menu_change('월별')

In [58]:
def set_monthly_searh_date_option_by_start_date(from_to):
    '''
    from_to = ['2025','09', '2025', '12']
    '''
    global driver

    # 2.1 st 년 
    Select(driver.find_element(By.ID,"sSearchYearFrom")).select_by_value(from_to[0])
    # 2.2 st 월
    Select(driver.find_element(By.ID,"sSearchMonthFrom")).select_by_value(from_to[1])
    time.sleep(2)

     # 2.3 end 년 
    Select(driver.find_element(By.ID,"sSearchYearTo")).select_by_value(from_to[2])
    # 2.4 end 월
    Select(driver.find_element(By.ID,"sSearchMonthTo")).select_by_value(from_to[3])


In [63]:
monthly_start_date_list = [
    # ['2024', '01', '2024', '02'],
    # ['2024', '03', '2024', '04'],
    # ['2024', '05', '2024', '06'],
    # ['2024', '07', '2024', '08'],
    # ['2024', '09', '2024', '10'],
    # ['2024', '11', '2024', '12'],
    # ['2025', '01', '2025', '02'],
    # ['2025', '03', '2025', '04'],
    # ['2025', '05', '2025', '06'],
    # ['2025', '07', '2025', '08'],
    # ['2025', '09', '2025', '10']
    # ['2024', '01', '2024', '02'],
    # ['2024', '07', '2024', '08'],
    # ['2025', '01', '2025', '02'],
    # ['2025', '07', '2025', '08']
    # ['2024', '01', '2024', '02'],

    # ['2024', '04', '2024', '05'],
    # ['2024', '07', '2024', '08'],
    ['2024', '10', '2024', '11'],
    ['2025', '01', '2025', '02'],
    ['2025', '04', '2025', '05'],
    ['2025', '07', '2025', '08'],
    ['2025', '10', '2025', '11']
]

#region
#지역 선택: 개별지역 돌아가며 -> 전체로 한다.
region_select = driver.find_element(By.ID, "sWideAreaCd")
region_option_list = region_select.find_elements(By.TAG_NAME, "option")
region_list = [r_opt.get_attribute("value") for r_opt in region_option_list if r_opt.get_attribute("value")]

# driver.set_page_load_timeout(240)#2분 지나면 타임아웃 에러 발생!

#date
date_len = len(monthly_start_date_list)
for di, start_date in enumerate(monthly_start_date_list):
    print(di+1, '/', date_len, start_date, end='')
    set_monthly_searh_date_option_by_start_date(start_date)
    time.sleep(0.5)

    for r_val in tqdm(region_list[:3]):#!! 일부만!!
        print(r_val)
        region_select = driver.find_element(By.ID, "sWideAreaCd")# !!이렇게 안쓰니 StaleElementReferenceException 에러가 나는데...
        Select(region_select).select_by_value(r_val)
        time.sleep( random.uniform(1,2) )

        # clickSearchButton() # type: ignore 
        # # time.sleep( random.uniform(1,2) )

        # downloadExcelFile()
        # # time.sleep( random.uniform(3,5) )

    # clickSearchButton() # type: ignore # time.sleep( random.uniform(5,8) )

    # downloadExcelFile()
    # time.sleep( random.uniform(3,5) )


1 / 5 ['2024', '10', '2024', '11']

  0%|          | 0/3 [00:00<?, ?it/s]

0105001


 33%|███▎      | 1/3 [00:01<00:03,  1.77s/it]

0105002


 67%|██████▋   | 2/3 [00:03<00:01,  1.65s/it]

0105003


100%|██████████| 3/3 [00:04<00:00,  1.48s/it]


2 / 5 ['2025', '01', '2025', '02']

  0%|          | 0/3 [00:00<?, ?it/s]

0105001


 33%|███▎      | 1/3 [00:01<00:02,  1.14s/it]

0105002


 67%|██████▋   | 2/3 [00:03<00:01,  1.63s/it]

0105003


100%|██████████| 3/3 [00:04<00:00,  1.54s/it]


3 / 5 ['2025', '04', '2025', '05']

  0%|          | 0/3 [00:00<?, ?it/s]

0105001


 33%|███▎      | 1/3 [00:01<00:03,  1.52s/it]

0105002


 67%|██████▋   | 2/3 [00:02<00:01,  1.31s/it]

0105003


100%|██████████| 3/3 [00:04<00:00,  1.39s/it]


4 / 5 ['2025', '07', '2025', '08']

  0%|          | 0/3 [00:00<?, ?it/s]

0105001


 33%|███▎      | 1/3 [00:01<00:03,  1.85s/it]

0105002


 67%|██████▋   | 2/3 [00:03<00:01,  1.65s/it]

0105003


100%|██████████| 3/3 [00:04<00:00,  1.60s/it]


5 / 5 ['2025', '10', '2025', '11']

  0%|          | 0/3 [00:00<?, ?it/s]

0105001


 33%|███▎      | 1/3 [00:01<00:03,  1.72s/it]

0105002


 67%|██████▋   | 2/3 [00:03<00:01,  1.75s/it]

0105003


100%|██████████| 3/3 [00:04<00:00,  1.62s/it]


1. 문제
1.1 Read timeout 발생 : 2분 넘어가니 짤 없이 짤림. 서비스 제공하는 서버에서 Time out 2분 설정한 것 같음.
 - 검색조건 범위를 줄여서 2분 안으로 설정.
 ! 지역단위로, 2개월 간격 조회 => 월 단위 조회는 1개월 더 추가되어 조회됨. 수동으로 6개월치 조회했을 때에도 7개월치 조회되었음.
 => 24.1~2, 24.4~5, 24.7~8, 24.10~11 
    25.1~2, 25.4~5, 25.7~8, 25.10~11  이렇게 조회한다!!

2.1 간헐적인 StaleElementReferenceException 발생 cf) https://spiaminto.tistory.com/19
 - 셀레니움 드라이버가 페이지의 요소를 미리 기록해두었으나 이후의 요청 시점에 그 요소를 DOM 에서 엑세스 할 수 없을때 발생하는 예외이다. 
 주로 WebDriverWait.until() 로 찾아놓은 element 객체에 element.getAttribute() 메서드 호출시점에 발생했다. headless 모드를 끄고 브라우저를 통해 직접 확인해 보았으나 DOM 에 변화가 없었음에도 불구하고 예외가 발생했다.
 => try except 걸어, 해당 element 다시 찾아서 -> click 하는 걸로 수정. 아직 재현 안되어서... 체크 필요!


'여기까지. 24.1~3 월 17개 지역 월단위 xls 파일로 다운로드 oK!'

자, 이렇게 다시!
    ['2024', '04', '2024', '05'],
    ['2024', '07', '2024', '08'],
    ['2024', '10', '2024', '11'],
    ['2025', '01', '2025', '02'],
    ['2025', '04', '2025', '05'],
    ['2025', '07', '2025', '08'],
    ['2025', '10', '2025', '11']


---------------------------------------------------------------------------
StaleElementReferenceException            Traceback (most recent call last)
Cell In[41], line 50
     46     time.sleep( random.uniform(1,2) )
     48     clickSearchButton() # type: ignore # time.sleep( random.uniform(5,8) )
---> 50     downloadExcelFile()
     51     time.sleep( random.uniform(3,5) )
     53 # clickSearchButton() # type: ignore # time.sleep( random.uniform(5,8) )
     54 
     55 # downloadExcelFile()
     56 # time.sleep( random.uniform(3,5) )

Cell In[38], line 9
      6 print('Excel 다운로드!')
      7 global driver
----> 9 exl_btn = driver.find_element(By.CSS_SELECTOR,".f_sch").find_element(By.TAG_NAME, "a")
     10 try:
     11     WebDriverWait(driver, 240).until(EC.element_to_be_clickable(exl_btn))

File ~/venv/eda_venv/lib/python3.12/site-packages/selenium/webdriver/remote/webelement.py:602, in WebElement.find_element(self, by, value)
    576 """Find an element given a By strategy and locator.
    577 
    578 Parameters:
   (...)    599     The first matching `WebElement` found on the page.

월 기간 검색조건 selenium 으로 조작했을때, 
 24.10 - 24.11 설정시 => 24.06 - 24.11로 설정되어 2분 넘게 소요됨!
 25.10 - 25.11 설정시 같은문제 발생 안함!

 ==> 자바스크립트에 callback 함수로 기간 6개월로 자동 설정하도록 되어있는 것 같음. 보통 그렇게 사용함 - 이유는 .. 여러가지인데 생략.
 

---------------------------------------------------------------------------
TimeoutError                              Traceback (most recent call last)
File ~/venv/eda_venv/lib/python3.12/site-packages/urllib3/connectionpool.py:534, in HTTPConnectionPool._make_request(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)
    533 try:
--> 534     response = conn.getresponse()
    535 except (BaseSSLError, OSError) as e:

File ~/venv/eda_venv/lib/python3.12/site-packages/urllib3/connection.py:565, in HTTPConnection.getresponse(self)
    564 # Get the response from http.client.HTTPConnection
--> 565 httplib_response = super().getresponse()
    567 try:

File /usr/lib/python3.12/http/client.py:1428, in HTTPConnection.getresponse(self)
   1427 try:
-> 1428     response.begin()
   1429 except ConnectionError:

File /usr/lib/python3.12/http/client.py:331, in HTTPResponse.begin(self)
    330 while True:
--> 331     version, status, reason = self._read_status()
    332     if status != CONTINUE:

File /usr/lib/python3.12/http/client.py:292, in HTTPResponse._read_status(self)
    291 def _read_status(self):
--> 292     line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
...
    369     ) from err
    371 # See the above comment about EAGAIN in Python 3.
    372 if hasattr(err, "errno") and err.errno in _blocking_errnos:

ReadTimeoutError: HTTPConnectionPool(host='localhost', port=59923): Read timed out. (read timeout=120)
Output is truncated. View as a scrollable element or open in a text editor. Adjust cell output settings...

In [10]:
driver.quit()